### PR TITLE
Bridge deployment fixes

### PR DESCRIPTION
**Pull request by kulmann**

### CODE REVIEW

**Review by IljaN — APPROVED** [review]

**Review by butonic — CHANGES_REQUESTED** [review]
@kulmann @IljaN tha bridge is supposed to enable oc10 deployments to switch to phoenix ... and it should be a starting point for a user by user migration. Also please verify and update the bridge docs: https://owncloud.github.io/ocis/deployment/bridge/

**Review by IljaN — CHANGES_REQUESTED** [review]
I think those are not templateable:
https://github.com/owncloud/ocis/pull/800/files#diff-c4194daa8f02ed28612366ca361d3b2f03fa1146e454710e12915f1601cf9fc9R6

Maybe this can be done by passing build-time vars to the Dockerfile

**Comment by IljaN** [comment]
> @kulmann @IljaN tha bridge is supposed to enable oc10 deployments to switch to phoenix ... and it should be a starting point for a user by user migration. Also please verify and update the bridge docs: https://owncloud.github.io/ocis/deployment/bridge/

@butonic Care to elaborate?

**Comment by IljaN** [comment]
Working on a solution to remove sed requirement on setup

**Comment by IljaN** [comment]
- [ ] Debug why ldap-login fails

**Comment by felixboehm** [comment]
the sed replacements should be executed at runtime, not buildtime.
And ideally we add the functionality to the ocis docker images and don't add a custom image here.

**Comment by IljaN** [comment]
- [ ] Change sed execution to runtime (how?)  https://github.com/owncloud/ocis/pull/800#issuecomment-724129021
- [ ] Change default domains to example.org (best practice)
- [ ] Split readme to production and local section

**Review by IljaN — APPROVED** [review]

**Review by jamu85 — APPROVED** [review]

**Review by micbar — APPROVED** [review]
LGTM

**Comment by butonic** [comment]
@kulmann how am I supposed to review this? I see commits from other PRs. Did a rebase go wrong? 36 files that changed?

The readme files in the deployment examples seems to be redundant with the docs. Get rid of one of the descriptions. We only want one source of truth.

**Review by butonic — CHANGES_REQUESTED** [review]
Rebase and deduplicate the documentation, please.

File: deployments/examples/ocis_oc10_backend/ocis/config/identifier-registration.yaml [hunk 1]
```diff
@@ -24,11 +24,11 @@ clients:
     insecure: yes
     trusted: yes
     redirect_uris:
-      - https://oc10.domain.com/apps/openidconnect/redirect/
-      - https://oc10.domain.com/apps/openidconnect/redirect
+      - http://oc10.example.org/apps/openidconnect/redirect
```
Comment (butonic) [inline deployments/examples/ocis_oc10_backend/ocis/config/identifier-registration.yaml:27]: for development purposes? if so add a comment that this must be removed in production instances!

File: deployments/examples/ocis_oc10_backend/ocis/config/identifier-registration.yaml [hunk 2]
```diff
@@ -24,11 +24,11 @@ clients:
...
+      - https://oc10.example.org/apps/openidconnect/redirect
     origins:
-      - http://oc10.domain.com
-      - https://oc10.domain.com
+      - http://oc10.example.org
```
Comment (butonic) [inline deployments/examples/ocis_oc10_backend/ocis/config/identifier-registration.yaml:30]: for development purposes? if so add a comment that this must be removed in production instances!

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 3]
```diff
@@ -9,8 +9,10 @@ geekdocFilePath: ocis_frontend_oc10_backend.md
 
 {{< toc >}}
 
-This deployment scenario shows how to use ocis as frontend for a existing owncloud 10 installation.
-ocis will allow owncloud 10 users to log in and work with their files.
+This deployment scenario shows how to use ocis as frontend for an existing ownCloud 10 installation.
+ocis will allow ownCloud 10 users to log in and work with their files. While the deployment scenario includes
+an ownCloud 10 instance, it only exists to show the necessary configuration for your already existing ownCloud 10
+installation.
```
Comment (butonic) [inline docs/ocis/deployment/ocis_oc10_backend.md:15]: so is this a development scenario? or a production scenario? if it can be used for both we need a section that describes what needs to be changed for a production deployment, like removing the http endpoints above.

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 4]
```diff
@@ -24,8 +26,8 @@ ocis will allow owncloud 10 users to log in and work with their files.
 
 ### Requirements
 
-* Server running Ubuntu 20.04 is public availible with a static ip address
-* Two A-records for both domains are pointing on the servers ip address
+* Server running Ubuntu 20.04 is public available with a static ip address
+* Two A-records for both domains are pointing to the servers ip address
```
Comment (butonic) [inline docs/ocis/deployment/ocis_oc10_backend.md:30]: maybe explain why? in order to prepare the user by user migration? running the two instances in parallel?

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 5]
```diff
@@ -184,15 +255,15 @@ clients:
...
     redirect_uris:
-      - http://ocis.domain.com/
-      - https://ocis.domain.com/
-      - http://ocis.domain.com/oidc-callback.html
-      - https://ocis.domain.com/oidc-callback.html
-      - http://ocis.domain.com/oidc-silent-redirect.html
-      - https://ocis.domain.com/oidc-silent-redirect.html
+      - http://ocis.example.org/
+      - https://ocis.example.org/
+      - http://ocis.example.org/oidc-callback.html
```
Comment (butonic) [inline docs/ocis/deployment/ocis_oc10_backend.md:260]: oidc requires https ... even if there are redirects on http ports they should not be listed as redirect_uris. the only exception is the desktop client

File: docs/ocis/deployment/ocis_oc10_backend.md (hunk 3 above)
Comment (butonic) [reply docs/ocis/deployment/ocis_oc10_backend.md:15]: Ok, so I misread some of the above. It nowhere mentions `development`. In any case maybe rephrase as
> This deployment scenario shows how to use ocis as frontend for an existing _productive_ ownCloud 10 installation.

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 6]
```diff
@@ -286,4 +360,22 @@ ADD apps/graphapi-0.1.0.tar.gz /var/www/owncloud/apps/
...
 The build is triggered by the terminal command `docker-compose build` from the root folder.
 
-Constraints: In this setup it's mandatory that the user has an email adress set in oc10.
+Constraints: In this setup it's mandatory that the user has an email address set and is assigned to at least one group in oc10.
+Especially the default admin user doesn't have an email assigned. If your admin user doesn't have an email address, yet, please
+set one: `docker-compose exec owncloud occ user:modify admin email "admin@example.org"`
+
+## Local deployment
+
+In case you want to start the bridge setup on your local development machine, there are a few steps necessary:
```
Comment (butonic) [inline docs/ocis/deployment/ocis_oc10_backend.md:369]: ok so this is a description of a production deployment and this section now tells the reader how to use it on the local machine. Or is this about a remote deployment vs a local deployment?

In any case I think it would make sense to clarify the deployment scenarios in `## Remote deployment` (for testing), `## Local deployment` (for development) and `## Production deployment`. Or maybe `## <type> scenario`?

**Review by butonic — COMMENTED** [review]
I added my comments ... and while I personally would like to see more comments in the used configuration files that explain WHY something is configured as it is, that can be done in a subsequent PR.

**Review by butonic — APPROVED** [review]
my previous review about updating the docs was addressed, so: :+1:

File: docs/ocis/deployment/ocis_oc10_backend.md (hunk 3 above)
Comment (phil-davis) [reply docs/ocis/deployment/ocis_oc10_backend.md:15]: And s/productive ownCloud 10 installation/production ownCloud 10 installation/

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 7]
```diff
@@ -24,8 +26,8 @@ ocis will allow owncloud 10 users to log in and work with their files.
 
 ### Requirements
 
-* Server running Ubuntu 20.04 is public availible with a static ip address
-* Two A-records for both domains are pointing on the servers ip address
+* Server running Ubuntu 20.04 is public available with a static ip address
```
Comment (phil-davis) [inline docs/ocis/deployment/ocis_oc10_backend.md:29]:
```suggestion
* Server running Ubuntu 20.04 is publicly available with a static ip address
```

File: docs/ocis/deployment/ocis_oc10_backend.md [hunk 8]
```diff
@@ -62,32 +64,28 @@ ocis will allow owncloud 10 users to log in and work with their files.
...
   `docker-compose up -d`
 
+  The domains from your `.env` will be used for building the configuration files during the docker start.
+
 ### Stack
 
-The application stack is separated in docker containers. One is a traefik proxy which is terminating ssl and forwards the https requests to the internal docker network. Additional, traefik is creating two certificates that are stored in the file `letsencrypt/acme.json` of the users home directory. In a local setup, this traefik is not included.
-The next container is the ocis server which is exposing the webservice on port 9200 to traefic and provides the oidc provider konnectd to owncloud.
+The application stack is separated in docker containers. One is a traefik proxy which is terminating ssl and forwards the https requests to the internal docker network. Additionally, traefik is creating two certificates that are stored in the file `letsencrypt/acme.json` of the users home directory. In a local setup, this traefik is not included.
+The next container is the ocis server which is exposing the webservice on port 9200 to traefic and provides the oidc provider `konnectd` to owncloud.
```
Comment (phil-davis) [inline docs/ocis/deployment/ocis_oc10_backend.md:88]:
```suggestion
The next container is the ocis server which is exposing the webservice on port 9200 to traefik and provides the oidc provider `konnectd` to owncloud.
```

File: docs/ocis/deployment/ocis_oc10_backend.md (hunk 6 above)
Comment (phil-davis) [inline docs/ocis/deployment/ocis_oc10_backend.md:369]:
```suggestion
If you want to start the bridge setup on your local development machine, there are a few steps necessary:
```

**Review by phil-davis — COMMENTED** [review]
Just some minor suggestions for word changes. Can be done in a later PR or ...

**Review by phil-davis — APPROVED** [review]

File: docs/ocis/deployment/ocis_oc10_backend.md (hunk 3 above)
Comment (IljaN) [reply docs/ocis/deployment/ocis_oc10_backend.md:15]: :heavy_check_mark:

**Comment by wkloucek** [comment]
I would propose to move the configuration of the linux servers into a separate section of the docs. Imho this is not that relevant for the examples itself and the exact steeps needed vary from setup to setup. The exact steeps may be only valid for ownCloud. Instead you should list common prerequisites like: docker and docker-compose must be installed and a dns record pointing to the public ip of the server.

Also this would deduplicate documentation as the setup of the linux server is contained in all three examples (ocis_external_konnectd, ocis_oc10_backend, ocis_traefik)

**Comment by IljaN** [comment]
- [  ] Squash

**Comment by IljaN** [comment]
> I would propose to move the configuration of the linux servers into a separate section of the docs. Imho this is not that relevant for the examples itself and the exact steeps needed vary from setup to setup. The exact steeps may be only valid for ownCloud's infrastructure. Instead you should list common prerequisites like: docker and docker-compose must be installed and a dns record pointing to the public ip of the server.
> 
> Also this would deduplicate documentation as the setup of the linux server is contained in all three examples (ocis_external_konnectd, ocis_oc10_backend, ocis_traefik)

:+1: Could you create a separate PR as soon as this is merged?

**Comment by butonic** [comment]
- [ ] add section about running a testsuite against this / a remote deployment
- [ ] delete moved docs/compose files from the docker-playground

**Comment by IljaN** [comment]
@butonic 
> * [ ]  add section about running a testsuite against this / a remote deployment

What's your idea here? Running both suites (ocis,10) after each other?  Just thinking that this might be a more generic topic not related to the bridge?

**Comment by butonic** [comment]
hmm https://drone.owncloud.com/owncloud/ocis/1394/16/7
```
@smokeTest @skipOnBruteForceProtection @issue-brute_force_protection-112 | 493s
-- | --
1551 | Scenario: send MKCOL requests to webDav endpoints without any authentication                                           # /srv/app/testrunner/tests/acceptance/features/apiAuthWebDav/webDavMKCOLAuth.feature:71 | 493s
1552 | When a user requests these endpoints with "MKCOL" with body "doesnotmatter" and no authentication about user "Alice" # AuthContext::userRequestsEndpointsWithBodyAndNoAuthThenStatusCodeAboutUser() | 501s
1553 | \| endpoint                                           \| | 501s
1554 | \| /remote.php/webdav/textfile0.txt                   \| | 501s
1555 | \| /remote.php/dav/files/%username%/textfile0.txt     \| | 501s
1556 | \| /remote.php/webdav/PARENT                          \| | 501s
1557 | \| /remote.php/dav/files/%username%/PARENT            \| | 501s
1558 | \| /remote.php/dav/files/%username%/PARENT/parent.txt \| | 501s
1559 | Then the HTTP status code of responses on all endpoints should be "401"                                              # FeatureContext::theHTTPStatusCodeOfResponsesOnAllEndpointsShouldBe() | 501s
1560 | │ | 502s
1561 | ╳  User 'Alice' should not exist but does exist | 502s
1562 | ╳  Failed asserting that true is false. | 502s
1563 | │ | 502s
1564 | └─ @AfterScenario # FeatureContext::afterScenario() | 502s
1565 |   | 502s
1566 | @api | 502s
1567 | Feature: MOVE file/folder | 502s
1568 |   | 502s
1569 | Background:                                                                              # /srv/app/testrunner/tests/acceptance/features/apiAuthWebDav/webDavMOVEAuth.feature:4 | 502s
1570 | Given user "Alice" has been created with default attributes and without skeleton files # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles() | 502s
1571 | Provisioning::usersHaveBeenCreatedUnexpected failure when creating a user: HTTP status 400 HTTP reason Bad Request OCS status 400 OCS message account already exists (Exception)
```

in the log I see
```
2020-11-12T09:16:08Z ERR could not delete user error="{\"id\":\"go.micro.client\",\"code\":500,\"detail\":\"service com.owncloud.api.accounts: not found\",\"status\":\"Internal Server Error\"}" service=ocs userid=Alice
--
57685 | 2020-11-12T09:16:08Z WRN basic auth enabled, use only for testing or development service=proxy
57686 | 2020-11-12T09:16:08Z INF access token is already provided pkg=rhttp service=storage
```

it could not look up the accounts service ... smells like out of file descriptors? maybe rebase on https://github.com/owncloud/ocis/pull/840 when it is merged?

**Review by butonic — APPROVED** [review]
only touches docs and compose files for example deployments. failing tests unrelated
